### ISSUE TITLE
Issue with the ActivityPub plugin

### ISSUE THREAD

**Issue by roytanck** [kovshenin/page-caching#36]
I'm testing the ActivityPub plugin (https://wordpress.org/plugins/activitypub/) on my blog. Sometimes I see the ActivityPub JSON instead of the regular page when visiting the site in a browser. This probably means that the JSON is cached and served to regular visitors. Is it possible to add a "variant" for this?

Steps to reproduce (assuming Surge is active with no custom configuration):

1. Install the ActivityPub plugin.
2. Configure at least one author profile in its settings.
3. Visit the Fediverse account (i.e. @user@example.com) using your favorite Mastodon client.
4. immediately after that visit the WordPress author page (example.com/author/user).

This does not happen to me every time, but the steps above should give you a decent chance.

Related: https://wordpress.org/support/topic/caching-conflict-between-wordpress-post-and-other-output/

**Comment by janboddez** [comment]
Not sure if still an issue, but https://dominikschilling.de/notes/http-accept-header-wordpress-cache-activitypub/ (as well as your blog, Roy) helped me figure this out.

TL;DR: it is possible to create a separate key/variant/... based on a request's Accept header.

My WordPress site would get hammered whenever I boosted one of my own posts. (None of the other caching solutions I'd tried would cache JSON.) Much less so with this in place; once traffic starts coming in, both HTML and JSON will be served (mostly) from cache, without the two formats getting mixed up!

**Comment by roytanck** [comment]
Just pretty much copy-pasted the solution from that blog post into my `cache-config.php`, and the issue appears to be gone. Thank you, @janboddez (and @ocean90).
